src/data/reports/glass-admin/GLASSDataMaintenanceDefaultRepository.ts

```diff
@@ -0,0 +1,499 @@
+    }
+
+    private async getCountriesOutsideNARegion(): Promise<string[]> {
+        const { organisationUnits: kosovoOu } = await this.api.metadata
+            .get({
+                organisationUnits: {
+                    fields: {
+                        id: true,
+                    },
+                    filter: {
+                        name: {
+                            eq: "Kosovo",
+                        },
+                        level: {
+                            eq: "3",
+                        },
+                        "parent.code": {
+                            eq: "NA",
+                        },
+                    },
+                },
+            })
+            .getData();
+
+        const { organisationUnits } = await this.api.metadata
+            .get({
+                organisationUnits: {
+                    fields: {
+                        id: true,
+                    },
+                    filter: {
+                        level: {
+                            eq: "3",
+                        },
+                        "parent.code": {
+                            ne: "NA",
+                        },
+                    },
+                    paging: false,
+                },
+            })
+            .getData();
+
+        return _.concat(organisationUnits, kosovoOu).map(orgUnit => orgUnit.id);
+    }
+
+    private uploadNewATC(year: string, atcItems: ATCItem[]): ATCItem[] {
+        const atcYears = atcItems.map(atcItem => _.parseInt(atcItem.year));
+        const previousVersionYear = Math.max(...atcYears);
+
+        const newItem: ATCItem = {
+            year: year,
+            version: "1",
+            uploadedDate: new Date().toISOString(),
+            currentVersion: _.parseInt(year) > previousVersionYear,
+            previousVersion: false,
+        };
+
+        return _(atcItems)
+            .map(atcItem => {
+                if (atcItem.currentVersion) {
+                    return {
+                        ...atcItem,
+                        currentVersion: false,
+                        previousVersion: true,
+                    };
+                } else if (atcItem.previousVersion) {
+                    return {
+                        ...atcItem,
+                        previousVersion: false,
+                    };
+                } else {
+                    return atcItem;
+                }
+            })
+            .concat(newItem)
+            .value();
+    }
+
+    private patchATCVersion(atcItems: ATCItem[], selectedItems: ATCItemIdentifier[]): ATCItem[] {
+        return atcItems.reduce((acc: ATCItem[], atcItem: ATCItem) => {
+            const matchingItem = selectedItems.find(
+                item => item.year === atcItem.year && _.parseInt(item.version) === _.parseInt(atcItem.version)
+            );
+
+            if (matchingItem) {
```

Comment:
We have forgotten more logic here:

1. If `matchingItem` is the `currentVersion`: we should change the new patched version to `currentVersion: true`, change `matchingItem` to `previousVersion: true` and look for the one with `previousVersion: true` to change it to `false` since it will no longer be `true`
2. If `matchingItem` is not the `currentVersion` and is not the `previousVersion`: what should we do? Will this new patched version be the one to be used in the calculations? the new patched version should now be the `currentVersion` (and therefore the previous `currentVersion` will be the `previousVersion` and the `previousVersion` will no longer be this)??
3. If `matchingItem` is the `previousVersion`: what should we do? Will this new patched version be the one to be used in the calculations? the new patched version should now be the `currentVersion` (and therefore the previous `currentVersion` will be the `previousVersion` and the `previousVersion` will no longer be this)??

Reply:
What makes sense to me it's the next logic:
1. If `matchingItem` is the `currentVersion`: we should change the new patched version to `currentVersion: true`, change `matchingItem` to `previousVersion: true` and look for the one with `previousVersion: true` to change it to `false` since it will no longer be `true`
2. Any other situation, simply add the new patched version with `previousVersion: false` and `currentVersion: false`